lambdas/data-migration2/src/granulesAndFiles.ts

```diff
@@ -266,14 +284,25 @@ export const migrateGranulesAndFiles = async (
 
   /* eslint-disable no-await-in-loop */
   while (record) {
-    // eslint-disable-next-line max-len
-    const result = await migrateGranuleAndFilesViaTransaction(record, migrationResult, knex, loggingInterval);
+    const result = await migrateGranuleAndFilesViaTransaction({
+      dynamoRecord: record,
+      granuleAndFileMigrationResult: migrationResult,
+      knex,
+      loggingInterval,
+      stream: errorFileWriteStream,
+    });
     migrationResult.granulesResult = result.granulesResult;
     migrationResult.filesResult = result.filesResult;
 
     await searchQueue.shift();
     record = await searchQueue.peek();
+
+    if (record) {
+      errorFileWriteStream.write(',\n');
+    }
   }
+  errorFileWriteStream.end('\n]}');
```

Comment:
Likewise, this could be a `closeErrorFileWriteStream` helper

Reply:
Do you mind expanding on why a helper to end the write stream is useful? It would only be one line, right? I am

Reply:
I ended up adding a end function to help ensure that the stream closes based on some feedback I got from @Jkovarik [`963d9f7` (#2179)](https://github.com/nasa/cumulus/pull/2179/commits/963d9f7adde0399981f6a5ecdb6b5ee44eca2086)